projects/packages/forms/src/blocks/field-select/edit.js

```diff
@@ -52,143 +47,48 @@ export default function DropdownFieldEdit( props ) {
 				'jetpack/input',
 				{ type: 'dropdown', placeholder: __( 'Select one option', 'jetpack-forms' ) },
 			],
+			[ 'jetpack/dropdown', { options, lock: { move: true, remove: true } } ],
 		];
-	}, [ required ] );
+	}, [ required, options ] );
 
 	const innerBlocksProps = useInnerBlocksProps(
 		{ className: 'jetpack-field-dropdown__wrapper' },
 		{
-			allowedBlocks: ALLOWED_INNER_BLOCKS,
+			allowedBlocks: [ ...ALLOWED_INNER_BLOCKS, 'jetpack/dropdown' ],
 			template,
 			templateLock: 'all',
 		}
 	);
-
+	/*
 	const optionWrapperStyles = {
 		className: 'jetpack-field-dropdown__popover',
 	};
 
 	const changeFocus = ( index, cursorToEnd ) =>
 		setFocus( optionsWrapper.current, '[role=textbox]', index, cursorToEnd );
-
-	const handleSingleValue = ( index, value ) => {
-		const _options = [ ...options ];
-
-		_options[ index ] = value;
-
-		setAttributes( { options: _options } );
-		changeFocus( index );
-	};
-
-	const handleMultiValues = ( index, array ) => {
-		const _options = [ ...attributes.options ];
-		const cursorToEnd = array[ array.length - 1 ] !== '';
-
-		if ( _options[ index ] ) {
-			_options[ index ] = array.shift();
-			index++;
-		}
-
-		_options.splice( index, 0, ...array );
-
-		setAttributes( { options: _options } );
-		changeFocus( index + array.length - 1, cursorToEnd );
-	};
-
-	const handleChangeOption = index => value => {
-		const values = ( value || '' ).split( '\n' ).filter( op => op && op.trim() !== '' );
-
-		if ( ! values.length ) {
-			return;
-		}
-
-		if ( values.length > 1 ) {
-			handleMultiValues( index, values );
-		} else {
-			handleSingleValue( index, values.pop() );
-		}
-	};
-
-	const handleKeyDown = index => e => {
-		// Create a new dropdown option when the user hits Enter.
-		// Previously handled with the onSplit prop, which was removed in https://github.com/WordPress/gutenberg/pull/54543
-		if ( 'Enter' !== e.key ) {
-			return;
-		}
-
-		e.preventDefault();
-
-		const value = attributes.options[ index ];
-
-		if ( ! value ) {
-			return;
-		}
-
-		const caretPos = getCaretPosition( e.target );
-		// splitValue is the value after the caret position when a user hits Enter
-		const splitValue = caretPos ? value.slice( caretPos ) : '';
-
-		handleMultiValues(
-			index,
-			splitValue ? [ value.slice( 0, caretPos ), splitValue ] : [ value, '' ]
-		);
-	};
-
-	const handleDeleteOption = index => () => {
-		if ( attributes.options.length === 1 ) {
-			return;
-		}
-
-		const _options = [ ...attributes.options ];
-		_options.splice( index, 1 );
-		if ( _options.length === 0 || _options.filter( option => option ).length === 0 ) {
-			return;
-		}
-		setAttributes( { options: _options } );
-		changeFocus( Math.max( index - 1, 0 ), true );
-	};
-
```

Comment:
So the select field now leverages dropdown and dropdown option blocks to store options. I was trying to understand backwards compat concerns. In testing, if I create a select dropdown, then update to this branch, my options disappear. So I think we need some kind of migration for existing blocks from the old options structure to the new? 

Not sure if something like [migrateOptionsToInnnerBlocks()](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/src/blocks/shared/deprecations/migrate-options-to-inner-blocks.js) could be adapted here.